src/main/java/de/blackforestsolutions/datamodel/Price.java

```diff
@@ -1,62 +1,91 @@
 package de.blackforestsolutions.datamodel;
 
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
 import lombok.Getter;
 import lombok.Setter;
 
 import java.io.Serializable;
+import java.math.BigDecimal;
 import java.util.Currency;
+import java.util.EnumMap;
 
 @Getter
 @JsonDeserialize(builder = Price.PriceBuilder.class)
 public final class Price implements Serializable {
 
     private static final long serialVersionUID = 6106269076155338045L;
 
-    private final double value;
+    @JsonPropertyOrder(alphabetic = true)
+    private final EnumMap<PriceCategory, BigDecimal> values;
 
     private final Currency currency;
 
     private final String symbol;
 
-    private final String affiliateLink;
+    @JsonPropertyOrder(alphabetic = true)
+    private final EnumMap<PriceCategory, String> affiliateLinks;
 
     /**
      * Copy constructor
      *
      * @param price you would like to copy
      */
     public Price(Price price) {
-        this.value = price.getValue();
+        this.values = price.getValues();
         this.currency = price.getCurrency();
         this.symbol = price.getSymbol();
-        this.affiliateLink = price.getAffiliateLink();
+        this.affiliateLinks = price.getAffiliateLinks();
     }
 
     private Price(PriceBuilder price) {
-        this.value = price.getValue();
+        this.values = price.getValues();
         this.currency = price.getCurrency();
         this.symbol = price.getSymbol();
-        this.affiliateLink = price.getAffiliateLink();
+        this.affiliateLinks = price.getAffiliateLinks();
+    }
+
+    public EnumMap<PriceCategory, BigDecimal> getValues() {
```

Comment:
Are these Values? For me it looks more like Categories, but onle one value is that the case s? <<if yes maybe we should rename the variable.